src/core_plugins/kibana/public/management/sections/indices/edit_index_pattern/edit_index_pattern.js

```diff
@@ -187,8 +226,18 @@ uiModules.get('apps/management')
     };
 
     $scope.$watch('fieldFilter', () => {
-      if ($scope.fieldFilter !== undefined && $state.tab === 'scriptedFields') {
-        updateScriptedFieldsTable($scope, $state);
+      if ($scope.fieldFilter !== undefined) {
+        if($state.tab === 'indexedFields') {
+          updateIndexedFieldsTable($scope, $state);
+        } else if($state.tab === 'scriptedFields') {
+          updateScriptedFieldsTable($scope, $state);
+        }
+      }
```

Comment:
Minor suggestion: how about early exiting here to avoid nesting, and a slightly terser switch instead of else if?

```js
if ($scope.fieldFilter === undefined) {
  return;
}

switch ($state.tab) {
  case 'indexedFields':
    updateIndexedFieldsTable($scope, $state);
  case 'scriptedFields':
    updateScriptedFieldsTable($scope, $state);
}
```

Reply:
Updated 👍